src/base/LruMap.h

```diff
@@ -148,39 +169,47 @@ LruMap<EntryValue, EntryCost>::add(const char *key, EntryValue *t)
     if (memLimit() == 0)
         return false;
 
-    index.push_front(new Entry(key, t));
+    auto e = new Entry(key, t, ttl);
+    memUsed_ += e->memCost() + e->key.length() + sizeof(Entry) + sizeof(MapPair);
+    index.push_front(e);
     storage.insert(MapPair(key, index.begin()));
 
     ++entries_;
+
+    // if over memory limit, start purging
+    while (memUsed_ > memLimit_)
```

Comment:
> Given that this code is an LRU

LruMap is a cache driven by an LRU policy. Obeying memory size limits (when enabled) is one of the guarantees that an LruMap object must provide. Thus, the answers to my questions are not obvious and correct answers are more likely to be "no" instead of "yes".

Your changes do not have to fix any old LruMap bugs, but they must not add new ones. In some cases, it is easier to fix the old bug than not to add a new one in the already buggy and confusing code, but it is your call.